framework/src/util/Utils.cpp

```diff
@@ -154,6 +154,13 @@ std::string GetCurrentWorkingDirectory()
   return std::string();
 }
 
+static const std::string s_CurrWorkingDir = InitCurrentWorkingDirectory();
```

Comment:
We need (from my understanding) to fulfill the following requirement here:

1. Get the current working directory in a thread-safe way

This affects all calls to the underlying OS provided functions, from any piece of code running in the current process. The only safe way to avoid synchronization issues I know of is to call such functions during static initialization of the library, which means constructing a static global. I would be happy to see another solution. Function local statics do not help because they cannot synchronize calls to the underlying OS function.